autogen/agentchat/groupchat.py

```diff
@@ -561,6 +562,39 @@ def _finalize_speaker(self, last_speaker: Agent, final: bool, name: str, agents:
         agent = self.agent_by_name(name)
         return agent if agent else self.next_agent(last_speaker, agents)
 
+    def _register_custom_model_clients(self, agent: ConversableAgent, selector: ConversableAgent):
+        for config in selector.llm_config['config_list']:
```

Comment:
`selector.llm_config['config_list']` doesn't necessarily exist.
A more fundamental question is, why do we need to depend on the `selector`?
An even more fundamental question is, since we only need the `llm_config` now from the `selector` to create the `speaker_selection_agent`, why not remove `selector` and just make `llm_config` an attribute of the group chat? The answer could be: that's a breaking change. Then, an alternative is to avoid creating a new `speaker_selection_agent` every time, but simply reuse the `selector` as the `speaker_selection_agent`. Then we don't need to worry about the client class registration in `GroupChat`.
@marklysze what do you think?

Reply:
The point you are raising:

> selector.llm_config['config_list'] doesn't necessarily exist.

is certainly valid but, after some attempts, I personally don't think that reusing the `selector` as the `speaker_selection_agent`, even resetting it, is a valid option. 

In my opinion passing the `llm_config` to the `GroupChat` is a much more straightforward and transparent option, that also allows devs to better understand that `GroupChat` uses agents behind the curtains and to further customize model configurations (even though this is probably a not very likely scenario which is why I relied on the `GroupChatManager`'s config)

Reply:
Some good points raised above. When I implemented the agent-based select speaker, I didn't consider the need to use a different configuration, hence the inclusion of the selector, which seemed like a logical place to get the config.

The `selector`, for the purposes of "auto" select speaker mode, is just to get the `llm_config` from the group chat manager. So, having a separate `llm_config` attribute on GroupChat that is clearly for the purpose of the "auto" select speaker agent chat would, I agree, make it simpler for a developer to understand.

Now, having said that...

> Then, an alternative is to avoid creating a new `speaker_selection_agent` every time, but simply reuse the `selector` as the `speaker_selection_agent`. Then we don't need to worry about the client class registration in `GroupChat`.
> @marklysze what do you think?

So, if I've understood this correctly, use the `selector`, which is the `GroupChatManager` object in place of the `speaker_selection_agent`. I'm not sure what the downstream affects would be of that. For example, the GroupChatManager has `run_chat` in the reply register, so would this be called during the select speaker agent-to-agent chat?

Reply:
Trying to use the `GroupChatManager` object as the selector I'm running into a recursion depth error which might be related to the `run_chat` method (but to be fair I didn't have a lot of time to debug and analyze the erorr in depth). 

Meanwhile i implemented the llm_config solution and fixed the issue with the `config_list` key with 2581fb16b390b665dc83c3374e6eae848595cc1d

Reply:
> @marklysze Trying to use the `GroupChatManager` object as the selector I'm running into a recursion depth error which might be related to the `run_chat` method (but to be fair I didn't have a lot of time to debug and analyze the erorr in depth).
> 
> Meanwhile i implemented the llm_config solution and fixed the issue with the `config_list` key with [2581fb1](https://github.com/microsoft/autogen/commit/2581fb16b390b665dc83c3374e6eae848595cc1d)

Hi @Matteo-Frattaroli, my feeling would be that it would be recursion as I believe that run_chat would be being called within itself if it was used within select speaker for a new chat.

Thanks, I can test with the new solution!